sqlglot/expressions.py

```diff
@@ -2844,6 +2844,14 @@ class WithSystemVersioningProperty(Property):
     }
 
 
+class WithReadOnlyProperty(Property):
+    arg_types = {"this": False}
+
+
+class WithCheckOptionProperty(Property):
+    arg_types = {"this": False}
```

Comment:
An alternative could to be to get rid of these two, reuse `QueryOption` and handle the cases as follows:

- Create a `Var` using `_parse_var_from_options` for `WITH READ ONLY` and `WITH CHECK OPTION` (so you'd want to set `raise_unmatched=False`).
- Handle `constraint` like you do now.

Only `QueryOption.this` will be needed to represent this. Generation can go in Oracle. I'm suggesting this because I'm not sure I like the property approach.

Reply:
Hmm, how about storing the kind (read only / check option) as a Var in `QueryOption.this` and the constraint in `QueryOption.expression`?

Reply:
That works too, I just didn't suggest it because the difference is minimal and it'll require making `this` optional.